### PR TITLE
fix rsync err when make release

### LINKED ISSUE #39513: fail to make quick-release (rsync error)
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):BUG REPORT

**Kubernetes version** (use `kubectl version`):Client Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.2850+4398e1edb00b7c-dirty", GitCommit:"4398e1edb00b7cd00b56abb9673467746063d148", GitTreeState:"dirty", BuildDate:"2017-01-06T09:06:04Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.2753+f95362f953b681-dirty", GitCommit:"f95362f953b681b495822e99bb772246312c25f9", GitTreeState:"dirty", BuildDate:"2017-01-04T08:46:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}

**What happened**:
```
# make quick-release
+++ [0106 16:59:51] Verifying Prerequisites....
+++ [0106 16:59:52] Building Docker image kube-build:build-9901a0c10c-4-v1.7.4-1
+++ [0106 16:59:56] Syncing sources to container
ERROR: password file must be owned by root when running as root
rsync error: syntax or usage error (code 1) at authenticate.c(200) [sender=3.1.1]
Makefile:361: recipe for target 'quick-release' failed
make: *** [quick-release] Error 1
```
**How to reproduce it** (as minimally and precisely as possible):

the owner of file is not right

### PR DESCRIPTION
**Pull request by xilabao**
fix #39513

### CODE REVIEW

**Comment by xilabao** [comment]
@jbeda PTAL

**Comment by xilabao** [comment]
@ixdy PTAL

File: build/common.sh [hunk 1]
```diff
@@ -407,6 +410,7 @@ function kube::build::clean() {
 # Set up the context directory for the kube-build image and build it.
 function kube::build::build_image() {
   mkdir -p "${LOCAL_OUTPUT_BUILD_CONTEXT}"
+  chown -R ${USER_ID}.${GROUP_ID} ${LOCAL_OUTPUT_BUILD_CONTEXT}
```
Comment (ixdy) [inline build/common.sh]: might be worth adding a comment as to why we're doing this.

also minor style nit - put vars in quotes (e.g. `"${LOCAL_OUTPUT_BUILD_CONTEXT}"`)

File: build/common.sh [hunk 2]
```diff
@@ -19,6 +19,9 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+USER_ID=`id -u`
```
Comment (ixdy) [inline build/common.sh]: nit: prefer `$(id -u)` to backticks.

**Review by ixdy — COMMENTED** [review]
basically looks good, just a few super minor suggestions.

**Comment by xilabao** [comment]
fixed. PTAL @ixdy

**Comment by ixdy** [comment]
@k8s-bot ok to test

This looks fine, but I'm still not sure how necessary it is.

**Comment by xilabao** [comment]
`make clean` can fix it. but `make clean` will remove _output directory.  other directories will be removed, such as `_output/local/` created by `hack/local-up-cluster.sh`. I think we can fix it without run `make clean` :)

**Comment by ixdy** [comment]
As I understand it, you created this PR because you can't run `make` as root without it.

In #39513 you explained that you needed to run `make` as root because the `.make` directory (and contents) were owned by root. These files would only be owned as root if you run `make` as root, though.

If you delete the `.make` directory and everything underneath it, `make` should again work as non-root.

You shouldn't ever need to run `make` as root.

**Comment by xilabao** [comment]
I understand what you mean. `hack/local-up-cluster.sh` suggest be run as root. and it may change the _output directory. If someone want to use the both command, he may run `make release` as root. If we make sure ${LOCAL_OUTPUT_BUILD_CONTEXT} owned by the right user, the issue in #39513 will not occur.

**Comment by ixdy** [comment]
ah, thanks, that was the piece of the puzzle I was missing.

it'd probably be best if `hack/local-up-cluster.sh` used `sudo` when necessary, rather than requiring running it as root, but that would be a more involved effort.

/lgtm
/approve

**Comment by roberthbailey** [comment]
I just tried doing a `make release` on my mac and this PR seems to have broken my ability to build code:

```shell
$ make quick-release
+++ [0126 12:12:04] Verifying Prerequisites....
+++ [0126 12:12:05] Using Docker for MacOS
chown: 146265.5000: illegal user name
!!! [0126 12:12:05] Call tree:
!!! [0126 12:12:05]  1: build/release.sh:35 kube::build::build_image(...)
make: *** [quick-release] Error 1
```

**Comment by ixdy** [comment]
It looks like the `.` between the uid and gid should be a `:`. I guess supporting `.` is a weird GNUism, even though the GNU manpage says to use `:`.

**Comment by roberthbailey** [comment]
I changed it to a `:` locally and can build again (previously I'd just commented out the chown line altogether). Do you have a PR out to fix this or should I send one?

**Pull request merged**